src/atlas/interpolation/method/structured/RegionalLinear2D.cc

```diff
@@ -0,0 +1,557 @@
+            indexI = sourceI;
+            colocatedX = false;
+          }
+        }
+        if (std::abs(targetX-sourceGrid.x(sourceNx-1)) < eps) {
+          indexI = sourceNx-1;
+          colocatedX = true;
+        }
+        const double targetY = targetPoints[targetJnode][1];
+        bool colocatedY = false;
+        int indexJ = -1;
+        for (size_t sourceJ = 0; sourceJ < sourceNy-1; ++sourceJ) {
+          if (std::abs(targetY-sourceGrid.y(sourceJ)) < eps) {
+            indexJ = sourceJ;
+            colocatedY = true;
+          }
+          if (reversedY) {
+            if (sourceGrid.y(sourceJ+1)+eps < targetY && targetY < sourceGrid.y(sourceJ)-eps) {
+              indexJ = sourceJ;
+              colocatedY = false;
+            }
+          } else {
+            if (sourceGrid.y(sourceJ)+eps < targetY && targetY < sourceGrid.y(sourceJ+1)-eps) {
+              indexJ = sourceJ;
+              colocatedY = false;
+            }
+          }
+        }
+        if (std::abs(targetY-sourceGrid.y(sourceNy-1)) < eps) {
+          indexJ = sourceNy-1;
+          colocatedY = true;
+        }
+
+        if (indexI == -1 || indexJ == -1) {
+          // Point outside of the domain, using nearest neighbor
+          if (indexI > -1) {
+            if (!colocatedX &&
+              (std::abs(targetX-sourceGrid.x(indexI+1)) < std::abs(targetX-sourceGrid.x(indexI)))) {
+              indexI += 1;
+            }
+          } else {
+            if (std::abs(targetX-sourceGrid.x(0)) < std::abs(targetX-sourceGrid.x(sourceNx-1))) {
+              indexI = 0;
+            } else {
+              indexI = sourceNx-1;
+            }
+          }
+          if (indexJ > -1) {
+            if (!colocatedY &&
+              (std::abs(targetY-sourceGrid.y(indexJ+1)) < std::abs(targetY-sourceGrid.y(indexJ)))) {
+              indexJ += 1;
+            }
+          } else {
+            if (std::abs(targetY-sourceGrid.y(0)) < std::abs(targetY-sourceGrid.y(sourceNy-1))) {
+              indexJ = 0;
+            } else {
+              indexJ = sourceNy-1;
+            }
+            std::cout << "WARNING: point outside of the domain" << std::endl;
+          }
+
+          // Colocated point (actually nearest neighbor)
+          colocatedX = true;
+          colocatedY = true;
+        }
+
+        // Bilinear interpolation factor
+        const double alphaX = 1.0-(sourceGrid.x(indexI)+sourceDx-targetX)/sourceDx;
+        const double alphaY = reversedY ? (sourceGrid.y(indexJ)-targetY)/sourceDy
+          : 1.0-(sourceGrid.y(indexJ)+sourceDy-targetY)/sourceDy;
+
+        // Points to find
+        std::vector<bool> toFind = {true, !colocatedX, !colocatedY, !colocatedX && !colocatedY};
+        std::vector<size_t> valueToFind = {indexI*sourceNy+indexJ, (indexI+1)*sourceNy+indexJ,
+          indexI*sourceNy+(indexJ+1), (indexI+1)*sourceNy+(indexJ+1)};
+        std::vector<int> foundIndex(4, -1);
```

Comment:
Since these are always size 4, it will be much better to use ```std::array<type,4>``` instead of `std::vector<type>`

Reply:
Done.